specifyweb/frontend/js_src/lib/components/WorkBench/WbUtils.tsx

```diff
@@ -6,35 +6,232 @@
+              aria-label={commonText.searchQuery()}
+              autoComplete="on"
+              placeholder={commonText.search()}
+              spellCheck
+              title={commonText.searchQuery()}
+              type="search"
+              onKeyDown={handleSearch}
+            />
+          </div>
+          {!isUploaded && hasPermission('/workbench/dataset', 'update') ? (
+            <div className="flex">
+              <Input.Text
+                forwardRef={replaceRef}
+                aria-label={wbText.replacementValue()}
+                autoComplete="on"
+                placeholder={wbText.replace()}
+                title={wbText.replacementValue()}
+                onKeyDown={handleReplace}
+              />
+            </div>
+          ) : undefined}
+          <span>
+            <WbAdvancedSearch
+              initialSearchPreferences={utils.searchPreferences}
+              onChange={(newSearchPreferences) => {
+                if (
+                  newSearchPreferences.navigation.direction !==
+                  utils.searchPreferences.navigation.direction
+                ) {
+                  // TODO: add workbench or cells to parent component
+                  // cells.flushIndexedCellData = true;
+                }
+                utils.searchPreferences = newSearchPreferences;
+                // TODO: figure out what searchCells with SettingsChange does
+                // if (utils.searchPreferences.search.liveUpdate)
+                //   utils.searchCells({
+                //     key: 'SettingsChange',
+                //   }).catch(softFail);
+              }}
+            />
+          </span>
+        </span>
+        <Navigation
+          label={wbText.searchResults()}
+          name="searchResults"
+          totalCount={cellCounts.searchResults}
+          utils={utils}
+          spreadsheetContainer={spreadsheetContainer}
+        />
+        {!isUploaded && hasPermission('/workbench/dataset', 'update') ? (
+          <Navigation
+            label={wbText.modifiedCells()}
+            name="modifiedCells"
+            totalCount={cellCounts.modifiedCells}
+            utils={utils}
+            spreadsheetContainer={spreadsheetContainer}
+          />
+        ) : undefined}
+        <Navigation
+          label={wbText.newCells()}
+          name="newCells"
+          totalCount={cellCounts.newCells}
+          utils={utils}
+          spreadsheetContainer={spreadsheetContainer}
+        />
+        {!isUploaded && (
+          <Navigation
+            label={wbText.errorCells()}
+            name="invalidCells"
+            totalCount={cellCounts.invalidCells}
+            utils={utils}
+            spreadsheetContainer={spreadsheetContainer}
+          />
+        )}
+      </div>
+    </>
+  );
+}
 
-// REFACTOR: rewrite to React
 /* eslint-disable functional/no-this-expression */
-export class WbUtils extends Backbone.View {
+export class WbUtils {
```

Comment:
can this class be dispanded in favor of `react states + utility functions` or is it still needed?
i.e `searchQuery` can be replaced with a React.useState
and `searchPreferences` can be replaced with `useCachedState` hook we have

Reply:
I think we can but there are some functions (and maybe even variables) in WbUtils that are used in other components/classes (eg: `CellMeta.ts`) through the Workbench object. I think further refactoring will basically boil down to figuring out how to decouple the objects inside the Workbench object.